Sources/Automerge/Document.swift

```diff
@@ -783,6 +783,24 @@ public final class Document: @unchecked Sendable {
         }
     }
 
+    /// Encode the Automerge document in a compressed binary format.
+    ///
+    /// - Parameters:
+    ///   - message: A message to attach to the auto-committed change (if any).
+    ///   - timestamp: The timestamp to attach to the auto-committed change (if any).
+    /// - Returns: The data that represents all the changes within this document.
+    ///
+    /// The `saveWithOptions` function also compacts the memory footprint of an Automerge document and increments the
+    /// result of ``heads()``, which indicates a specific point in time for the history of the document.
+    public func saveWithOptions(message: String, timestamp: Date) -> Data {
+        sync {
+            self.doc.wrapErrors {
+                sendObjectWillChange()
+                return Data($0.saveWithOptions(msg: message, time: Int64(timestamp.timeIntervalSince1970)))
```

Comment:
Will make `message` optional since that can be null on the underlying Rust type. I think giving the timestamp a default `Date().timeIntervalSince1970` value is the most convenient.